sycl/doc/SpecializationConstants.md

```diff
@@ -200,93 +200,56 @@ and the user says
  - The SpecConstants pass in the post-link will have the following IR as input (`sret` conversion is omitted for clarity):
 
 ```
-  %spec_const = call %struct.POD __sycl_getCompositeSpecConstantValue<POD type mangling> ("MyConst_mangled")
-```
+%struct.POD = type { [2 x %struct.A], i32 }
+%struct.A = type { i32, float }
 
-where `__sycl_getCompositeSpecConstantValue` is a new "intrinsic"
- (in addition to `__sycl_getSpecConstantValue`) recognized by SpecConstants pass,
- which creates a value of a composite (of non-primitive type) specialization constant.
- It does not need a default value, because its default value consists of default
- valued of its leaf specialization constants (see below).
+%spec_const = call %struct.POD __sycl_getSpecConstantValue<POD type mangling> ("MyConst_mangled")
+```
 
- - after spec constant enumeration (symbolic -\> int ID translation), the SpecConstants pass
- will handle the `__sycl_getCompositeSpecConstantValue`. Given the knowledge of the composite
- specialization constant's type (`%struct.POD`), the pass will traverse its leaf
- fields and generate 5 "primitive" spec constants using already existing SPIR-V intrinsic:
+Based on the fact that `__sycl_getSpecConstantValue` returns `llvm::StructType`,
```

Comment:
handling of `__sycl_getCompositeSpecConstantValue` is significantly different from handling of `__sycl_getSpecConstantValue`, that's why I designed it this way. I don't think overloading brings extra clarity here

Reply:
Returned this intrinsic back in e40b404e10c2bad9b4b40bf501ff6cd2bb17b2f4